php/libraries/Database.class.inc

```diff
@@ -302,7 +305,7 @@ class Database
 
         if (DEBUG) {
             fwrite(STDERR, $query."\n");
-            return;
+            throw new DatabaseException("Exception due to debug mode");
```

Comment:
This doesn't make sense to be an exception instead of a return. It would mean that any page with an insert would crash in debug mode.